src/llmq/quorums_chainlocks.cpp

```diff
@@ -475,6 +475,19 @@ bool CChainLocksHandler::IsTxSafeForMining(const uint256& txid)
         }
     }
 
+    // Fallback to mempool in unexpected cases
+    if (txAge == -1) {
+        // Not found? Let's check mempool
+        auto info = mempool.info(txid);
+        if (!info.tx) {
+            // Not in mempool either, this should not really happen
+            return false;
+        }
+        LOCK(cs);
+        txFirstSeenTime.emplace(txid, info.nTime);
```

Comment:
Hmm, I really don't like IsXXX methods with side effects. I'd prefer to not have this commit at all to be honest. Instead, we should maybe just do this on v0.14.0.x:
```diff
diff --git a/src/llmq/quorums_chainlocks.cpp b/src/llmq/quorums_chainlocks.cpp
index cdb5d99ac1..221b8d0a9c 100644
--- a/src/llmq/quorums_chainlocks.cpp
+++ b/src/llmq/quorums_chainlocks.cpp
@@ -347,10 +347,6 @@ void CChainLocksHandler::TrySignChainTip()
 
 void CChainLocksHandler::SyncTransaction(const CTransaction& tx, const CBlockIndex* pindex, int posInBlock)
 {
-    if (!masternodeSync.IsBlockchainSynced()) {
-        return;
-    }
-
     bool handleTx = true;
     if (tx.IsCoinBase() || tx.vin.empty()) {
         handleTx = false;
@@ -363,6 +359,10 @@ void CChainLocksHandler::SyncTransaction(const CTransaction& tx, const CBlockInd
         txFirstSeenTime.emplace(tx.GetHash(), curTime);
     }
 
+    if (!masternodeSync.IsBlockchainSynced()) {
+        return;
+    }
+
     // We listen for SyncTransaction so that we can collect all TX ids of all included TXs of newly received blocks
     // We need this information later when we try to sign a new tip, so that we can determine if all included TXs are
     // safe.
```

It'll cause TXs to linger in the mempool for 10 minutes, but that's still a lot better then never mining them.

Reply:
Makes sense. Dropped this commit.